src/ngx_http_lua_regex.c

```diff
@@ -363,6 +365,7 @@ ngx_http_lua_ngx_re_match_helper(lua_State *L, int wantcaps)
         old_pool = ngx_http_lua_pcre_malloc_init(pool);
 
         sd = pcre_study(re_comp.regex, PCRE_STUDY_JIT_COMPILE, &msg);
+        pcre_assign_jit_stack(sd, NULL, lmcf->jit_stack);
```

Comment:
Maybe we can avoid this C function call when the user has never set any jit stack size explicitly? We could make `lmcf->jit_stack` default to `NULL` and always tests the nullity of `lmcf->jit_stack` before calling `pcre_assign_jit_stack`. Thoughts?